js/customize-snapshots.js

```diff
@@ -731,16 +701,99 @@
 		 * @returns {string} URL.
 		 */
 		getSnapshotFrontendPreviewUrl: function getSnapshotFrontendPreviewUrl() {
-			var snapshot = this, a = document.createElement( 'a' );
+			var snapshot = this, a = document.createElement( 'a' ),
+				params = {};
+
+			if ( api.settings.changeset && api.settings.changeset.uuid ) {
+				params.customize_changeset_uuid = api.settings.changeset.uuid;
+			} else {
+				params.customize_changeset_uuid = snapshot.data.uuid;
+			}
+
 			a.href = snapshot.frontendPreviewUrl.get();
-			if ( a.search ) {
-				a.search += '&';
+			if ( snapshot.statusButton.disbleButton.get() ) {
+				return a.href;
```

Comment:
@westonruter / @miina Why are we returning just the `a.href` when the button is not enabled?

Reply:
Which means if the changeset is saved as draft/future/pending, we will just show the frontend without the `customize_changeset_uuid` param which will not reflect the changes on frontend.

Reply:
Yes, I think that's right.